src/stm32f1/adc.cpp

```diff
@@ -301,19 +320,18 @@ float adc_peripheral_manager::read_channel(pins p_pin)
   // Lock the lock.
   std::lock_guard<hal::basic_lock> acquire_lock(*m_lock);
 
-  // NOLINTNEXTLINE(performance-no-int-to-ptr)
-  auto adc_reg = reinterpret_cast<adc_reg_t*>(adc_reg_location);
+  auto& adc_reg = to_reg(adc_reg_location);
   // Set the specified channel to be sampled.
-  hal::bit_modify(adc_reg->regular_sequence_3)
+  hal::bit_modify(adc_reg.regular_sequence_3)
     .insert<adc_regular_sequence_register_3::first_conversion>(
       hal::value(p_pin));
 
   // Start adc conversion.
-  hal::bit_modify(adc_reg->control_2)
+  hal::bit_modify(adc_reg.control_2)
     .set<adc_control_register_2::ad_converter_on>();
 
   // Wait for conversion to complete.
-  while (bit_extract<adc_status_register::end_of_conversion>(adc_reg->status) ==
+  while (bit_extract<adc_status_register::end_of_conversion>(adc_reg.status) ==
          0) {
```

Comment:
This is a nitpick but I think this would look better written as:
```suggestion
  while (not bit_extract<adc_status_register::end_of_conversion>(adc_reg.status)) {
```

Reads like "not the end of conversion".